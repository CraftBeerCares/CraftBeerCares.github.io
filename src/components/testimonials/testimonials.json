[
  {
    "quote": "Craft Beer Cares were one of the first UK groups to create a space for craft beer lovers to drink for a cause, and their charitable work is hugely important to the industry and the organisations they support. G knows how to throw a great party, and I'm looking forward to attending more CBC fundraisers in the future!",
    "author": "Ruvani de Silva"
  },
  {
    "quote": "Craft Beer Cares tightly knits the best aspects of the beer industry: excellent beer and excellent community in a super fun festival format that also does some good. What is there not to love about that?",
    "author": "Natasha W."
  },
  {
    "quote": "I'm a CBC veteran volunteer - And I can't wait for the next festival. It's the friendliest, fun-est crowd and the volunteers are an amazing bunch of enthusiastic, knowledgeable, efficient and kind people who make the whole event an all day party.",
    "author": "Mona B."
  },
  {
    "quote": "What impressed me most was the collegial and friendly environment, among the staff but also between us and the punters, who all seemed to understand that this wasn't your usual festival.",
    "author": "Ewan M."
  },
  {
    "quote": "Had great fun attending the festival for the first time some years ago when it was held at Brew Club and such an honourable experience having been able to bring in my swissrolls at another festival in 2019!",
    "author": "Shima"
  }
]
