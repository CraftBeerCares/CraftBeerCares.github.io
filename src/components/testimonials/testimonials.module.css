.testimonial {
  width: 420px;
  padding-left: 3.4rem;
}

.dotList {
  position: relative;
  margin-top: 0.5rem;
  margin-bottom: 1.5rem;
}
