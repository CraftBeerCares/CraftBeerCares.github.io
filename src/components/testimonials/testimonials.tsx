"use client";

import Quote from "../quote/quote";
import testimonials from "./testimonials.json";
import {
  CarouselProvider,
  Slider,
  Slide,
  ButtonBack,
  ButtonNext,
  DotGroup,
} from "pure-react-carousel";
import styles from "./testimonials.module.css";
import "pure-react-carousel/dist/react-carousel.es.css";
import { useEffect, useRef, useState } from "react";

export const getQuoteNumber = (div: HTMLDivElement | null) : number => {
  return window.matchMedia("(min-width: 701px)").matches
    ? Math.min(Math.floor((div?.offsetWidth ?? window.innerWidth) / 420), 3)
    : 1;
}

function Testimonials() {
  const elementRef = useRef<HTMLDivElement>(null);

  const [quoteNumber, setQuoteNumber] = useState(1);

  const setQuoteNumberEffect = () => {
    setQuoteNumber(getQuoteNumber(elementRef.current));
  };

  // listen for resize events
  useEffect(() => {
    setQuoteNumberEffect();

    window.addEventListener("resize", setQuoteNumberEffect);
    return () => {
      window.removeEventListener("resize", setQuoteNumberEffect);
    };
  }, []);

  return (
    <CarouselProvider
      naturalSlideHeight={105}
      naturalSlideWidth={102}
      totalSlides={testimonials.length}
      visibleSlides={quoteNumber}
      isIntrinsicHeight={true}
      infinite={true}
    >
      <Slider>
        {testimonials.map(({ quote, author }, index) => (
          <Slide index={index} key={index} className={styles.quoteBox}>
            <Quote quote={quote} author={author} />
          </Slide>
        ))}
      </Slider>
      <div className={styles.controls} ref={elementRef}>
        <ButtonBack className={styles.button}>Back</ButtonBack>
        <DotGroup className={styles.dots} />
        <ButtonNext className={styles.button}>Next</ButtonNext>
      </div>
    </CarouselProvider>
  );
}

export default Testimonials;
