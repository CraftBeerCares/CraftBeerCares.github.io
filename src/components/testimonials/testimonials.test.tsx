import { render } from "@testing-library/react";
import { Testimonials } from "./testimonials";

it("renders Testimonial", () => {
  const { container } = render(<Testimonials />);
  expect(container).toMatchSnapshot();
});
