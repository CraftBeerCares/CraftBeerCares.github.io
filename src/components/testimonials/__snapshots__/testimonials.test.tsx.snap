// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Testimonial 1`] = `
<div>
  <div
    class="testimonials"
  >
    <div
      class="react-multi-carousel-list  testimonials"
      dir="ltr"
    >
      <ul
        class="react-multi-carousel-track "
        style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
      />
    </div>
  </div>
</div>
`;
